core/src/services/iframe.js

```diff
@@ -347,6 +347,8 @@ class IframeClass {
       config.iframe['vg'] = this.canCache(componentData.viewGroup)
         ? componentData.viewGroup
         : undefined;
+      config.iframe.luigi.currentNode = componentData.currentNode;
+      const internalData = await component.prepareInternalData(config);
```

Comment:
Would it be possible (if it is worth it) to remove the await (and async from function declaration) changing the method invocation:

 const message = {
        msg: 'luigi.navigate',
        viewUrl: viewUrl,
        context: JSON.stringify(
          Object.assign({}, componentData.context, { goBackContext })
        ),
        nodeParams: JSON.stringify(Object.assign({}, componentData.nodeParams)),
        pathParams: JSON.stringify(Object.assign({}, componentData.pathParams)),
        internal: JSON.stringify(internalData)
      };

      const withSync = componentData.isNavigationSyncEnabled;
      if (withSync) {
        // default, send navigation event to client
        IframeHelpers.sendMessageToIframe(config.iframe, message);
        this.setOkResponseHandler(config, component, node);
      } else {
        // `withoutSync()` used. client navigation was skipped, reset after one-time use.
        component.set({ isNavigationSyncEnabled: true });
      }

      // clear goBackContext and reset navigateBack after sending it to the client
      component.set({ goBackContext: undefined, isNavigateBack: false });

We will just lose control on when sending message to the Iframe.... (that I guess it would not be a problem...

Reply:
@legteodav Thanks for your time taking a look and provide a good solution. I will play this solution around and see how it works.